Documentation/design-docs/AssemblyLoadContext.Activate.md

```diff
@@ -0,0 +1,306 @@
+  }
+}
+```
+
+`ActiveContext` is stored in `AsyncLocal<AssemblyLoadContext>`. It represents the most recent `Activated()` `AssemblyLoadContext`.
+
+
+`Activate()` is intended to be used in a using block and/or statement.
+
+When `Activate()` is called it make the activated `AssemblyLoadContext` the current executing assembly.  This will impact the preference order for loading assemblies and resolving type names.
+
+Before `Activate()` is called, the behavior of the exitisting API's will remanin unchanged.  The `AssemblyLoadContext` containing the API caller will determine the preferred `AssemblyLaodContext`.
+
+Once `Activate()` is called, the associated `AssemblyLoadContext` will become the preferred `AssemblyLoadContext`.  The value will be reflected in `ActiveContext`
+* This preference will affect the current thread and its asynchronous control flow.
+* The behavior will persist in the current thread until the returned `IDisposable` is disposed.
+* **or**  until a nested call supercedes the preference.
+* `IDisposable.Dispose()` will restore the previous `ActiveContext` to support nesting.
+* Async workers will inherit their parent's ActiveContext when they are created. Their behavior will not be affected when `IDisposable.Dispose()` is called in the parent thread.  The worker can call `Activate()` to modifiy its own `ActiveContext`.
+* `Thread`s and `ThreadPool` queued work items should also be affected
+
+**TBD** naming.  Want a method name that makes sense with using. Want property getter name to be associated with method name. `Activate` and `ActiveContext` were chosen because the verb and noun are logically related in English.
+
+## Sample Usage
+### Basic Usage
+```C#
+AssemblyLoadContext alc = new AssemblyLoadContext();
+using (alc.Activate())
+{
+  // In this block, alc acts as the current executing context.
+  //
+  // This does not mean all assemblies will be loaded into it. Nor does it mean the set of type
+  // resolved by it will come exclusively from assemblies it has loaded.
+  //
+  // Rather when alc is the ActiveContext
+  //
+  // When loading assemblies, alc.Load() is called first, before falling back to default.
+  // alc.Resolve(...) handler will be called before AppDomain.ResolvingAssembly(...)
+  //
+  // When resolving types, the type cache in this alc is consulted. It will include types
+  // loaded into other AssemblyLoadContexts but used by this context
+
+  Assembly assembly = Assembly.Load(myPlugin); // Attempts to loads calling alc.Load() first, with normal fallback.
+
+  Type t = GetType("MyPlugin.MyType"); // Resolves type as if alc was active execution context
+
+  // Affects async methods
+  TResult result = await AsyncMethod();
+
+  // Ideally also affects threads and queued work items.
+  Thread t = new Thread(new ThreadStart(ThreadProc));
+  t.Start();
+
+  ThreadPool.QueueUserWorkItem(ThreadProc);
+}
+```
+Similarly
+```C#
+Assembly myAssembly = Assembly.LoadFrom(path);
```

Comment:
Two thing are on play here:
* I was thinking about this design in the sense of virtualization, so I was sometimes considering `Assembly.LoadFrom(path)` loading into the `ActiveContext`
* I was working offline and I could remember which APIcreated a new ALC: `Assembly.LoadFrom(path)` or `Assembly.LoadFile(path)`

Given where this landed, I will change to `LoadFile(path)` because modifying LoadFrom()` does not feel necessary.